docs/tutorials/basic/dart.md

```diff
@@ -0,0 +1,540 @@
+---
+bodyclass: docs
+layout: docs
+headline: gRPC Basics - Dart
+sidenav: doc-side-tutorial-nav.html
+type: markdown
+---
+<p class="lead">This tutorial provides a basic Dart programmer's introduction to
+working with gRPC.</p>
+
+By walking through this example you'll learn how to:
+
+- Define a service in a .proto file.
+- Generate server and client code using the protocol buffer compiler.
+- Use the Dart gRPC API to write a simple client and server for your service.
+
+It assumes that you have read the [Overview](/docs/index.html) and are familiar
+with [protocol buffers](https://developers.google.com/protocol-buffers/docs/overview). Note that the
+example in this tutorial uses the proto3 version of the protocol buffers
+language, which is currently in beta release: you can find out more in the
+[proto3 language
+guide](https://developers.google.com/protocol-buffers/docs/proto3),
+and see the [release notes](https://github.com/google/protobuf/releases) for the
+new version in the protocol buffers Github repository.
+
+<div id="toc"></div>
+
+## Why use gRPC?
+
+Our example is a simple route mapping application that lets clients get
+information about features on their route, create a summary of their route, and
+exchange route information such as traffic updates with the server and other
+clients.
+
+With gRPC we can define our service once in a .proto file and implement clients
+and servers in any of gRPC's supported languages, which in turn can be run in
+environments ranging from servers inside Google to your own tablet - all the
+complexity of communication between different languages and environments is
+handled for you by gRPC. We also get all the advantages of working with protocol
+buffers, including efficient serialization, a simple IDL, and easy interface
+updating.
+
+## Example code and setup
+
+The example code for our tutorial is in
+[grpc/grpc-dart/example/route_guide](https://github.com/grpc/grpc-dart/tree/master/example/route_guide).
+To download the example, clone the `grpc-dart` repository by running the following
+command:
+
+```
+$ git clone https://github.com/grpc/grpc-dart.git
+```
+
+Then change your current directory to `grpc-dart/example/route_guide`:
+
+```
+$ cd grpc-dart/example/route_guide
```

Comment:
are we missing `pub get` ?

Reply:
I have it down in the 'Try it out!' section.